pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -48,6 +51,53 @@ func (nc *nodeConfig) Configure() error {
 	if err := nc.handleCSRs(); err != nil {
 		return errors.Wrap(err, "handling CSR for the given node failed")
 	}
+
+	var instanceID string
+	// As the CSR has been approved get the Kubernetes node object associated
+	// the Windows VM created
+	if nc.Windows.GetCredentials() != nil && len(nc.Windows.GetCredentials().GetInstanceId()) > 0 {
```

Comment:
Can't you check for `nc.Windows.GetCredentials().GetInstanceId() != ""` instead of the `len`?